### DISCUSSION TITLE
Coloring achievement type indicators for easier visual parsing

### DISCUSSION THREAD

**Discussion by hb-robo** [RetroAchievements/RAWeb#2146] [Ideas]
(Prompted by a discussion on the #retroachievements channel on the RA Discord.)

A [new set for Beyond Good & Evil](https://retroachievements.org/game/3061) seems to have caused some minor complaints about the visual indicators for achievement types. The set has 115 achievements, with 1 Win Condition, 17 Progression, and 97 Missable. Because every single achievement has a type indicator, a few folks complained that it is difficult to visually scan for the Progression-related elements in the set.

One proposed suggestion was for the set to simply abandon the Missable type altogether, but these are not equivalent concepts in terms of information given to the player, and an unsuspecting player would rightfully be upset at the lack of marks. This also opens up the somewhat slippery slope of allowing developers to choose when to add the Missable type, which erodes confidence in the types to begin with.

A superior solution in my mind would be to simply color-code the type indicators. This restores visual parsing and avoids putting pressure on achievement developers to alter set design for the sake of what is essentially a front-end complaint.

Here is a simple mock-up using some examples from the Beyond Good and Evil set:

![image](https://github.com/RetroAchievements/RAWeb/assets/26484694/bc7655ed-f3dc-4fee-b591-74381c1843e1)

**Comment by wescopeland** [comment]
I have a different perspective. I don't believe this is primarily a front-end issue. If that were the case, concerns would have likely surfaced sooner than before this set was published. Instead, this seems like an overclassification issue.

Every non-beaten-related achievement in this set is marked as missable. This diminishes the effectiveness of the categorization that the types system provides, and introduces ambiguity in guiding players to understand and prioritize achievements in the set.

> This also opens up the somewhat slippery slope of allowing developers to choose when to add the Missable type, which erodes confidence in the types to begin with.

Adding missable types is not currently a requirement for developers: https://docs.retroachievements.org/Achievement-Set-Requirements/#recommended-but-not-required

**Reply by hb-robo** [reply]
I think all of the stances you hold are reasonable. However, bear with me and allow me to prod a little:

> Adding missable types is not currently a requirement for developers: https://docs.retroachievements.org/Achievement-Set-Requirements/#recommended-but-not-required

I probably shouldn't have brought set design enforcement into the original post, unnecessarily politicized the discussion. But while I was aware that was the protocol prior to the recent Missable rework, I thought that this was now a standard expectation for devs, similar to how Progression must be marked. Maybe some of that will be reworked at some point, but I was wrong on that, so let's move back to the main topic.

> I don't believe this is primarily a front-end issue. If that were the case, concerns would have likely surfaced sooner than before this set was published. Instead, this seems like an overclassification issue.

Beyond Good and Evil isn't unique in its overuse of the missable mark, but it is the first such notable case that's cropped up since the Missable type indicator was moved from the old [m] nomenclature. There are older sets that people also believe over-use the Missable mark, and those complaints about the design logic of the set are fair to acknowledge.

However, I want to stress that this isn't an issue of a set having too many Missables in isolation. The problem is that the Missable mark now shares the same visual space as the Progression-related indicators, with the same silhouette and color profile, and people are having trouble visually distinguishing them from one another as they scroll through the set. This would be true in any block of consecutively marked achievements, even if the block ultimately is a small portion of a wider set that does not abuse the Missable categorization, like in our example.

The core user story is "I can't easily see the progression achievements when there are missables surrounding them." This was not something that I anecdotally ever saw discussed before the Missable tag was moved to the same space - and why would it, when the two visually appeared completely differently? Passing the buck to DevComp to prevent this complaint from occurring going forward seems silly when you can implement a change that maintains indicator parsability regardless of how the set is constructed. If you find the colors tacky (or possibly they aren't A11Y-compliant), perhaps a checkbox to filter down to Progression only could be helpful, similar to the one in place for missable achievements.

Anyway, that's my $0.02 on that. I have a ton of respect for your work on this so far but I find it to be low-hanging fruit.
